### ISSUE TITLE
Logs button causes app to freeze

### ISSUE THREAD

**Issue by SilmarillionWorshipper** [modrinth/code#1127]
### Describe the bug

When attempting to view the logs for any of the mod profiles, the app will freeze and refuse to close unless killed.

### Steps to reproduce

1. Go to any mod profile
2. Click on the logs button

### System information

Linux (Pop!_OS 22.04)

**Comment by jcv99** [comment]
Same on fedora 39 with appimage.
Operating System: Fedora Linux 39
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.12
Kernel Version: 6.8.7-200.fc39.x86_64 (64-bit)
Graphics Platform: X11
Processors: 16 × Intel® Core™ i7-10700KF CPU @ 3.80GHz
Memory: 15,5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3070/PCIe/SSE2
Manufacturer: ASUS

**Comment by quaap** [comment]
Same here.  Manjaro/Arch with XFCE Kernel 6.9.0.

**Comment by Sl3epy** [comment]
Same here. On Ubuntu and Gnome

**Comment by AstroOrbis** [comment]
Same here with Arch Linux (which I use btw) on Plasma and Gnome with both X11 and Wayland (latest on all)

**Comment by CommanderGL** [comment]
Has this been seen?

**Comment by LapisPhoenix** [comment]
Same here, it will just hang there and do nothing, ive tried several ways of killing it but nothing seems to work, other than restarting.

Linux Mint 21.3 Cinnamon
![prewview](https://github.com/modrinth/theseus/assets/88260244/849fdbc6-60a6-424f-b3d1-2cba09624540)

**Comment by CommanderGL** [comment]
@LapisPhoenix I can quit by running `kill-all modrinth-app` on Arch but don't know about Mint.

**Comment by Norbiros** [comment]
The same bug was mentioned here: https://github.com/modrinth/theseus/issues/1216#issuecomment-2158897849

**Comment by LapisPhoenix** [comment]
> @LapisPhoenix I can quit by running `kill-all modrinth-app` on Arch but don't know about Mint.

Sorry for late reply, thanks for letting me know. On Mint I need to do `killall modrinth-app` and it works. Thanks for the work around.

**Comment by CommanderGL** [comment]
@LapisPhoenix sorry yeah I meant `killall modrinth-app` the `-` was a mistake. I'll update my reply.

**Comment by Norbiros** [comment]
This bug was reintroduced in https://github.com/modrinth/code/commit/c84e2e0537517f9e9f06b2b1eb5c0c6e2011171a

**Comment by Call-Me-Alex** [comment]
@Norbiros can confirm

**Comment by okunamayanad** [comment]
Ubuntu 24.04, opening any modpack's logs page makes the app freeze and turns the window to gray. here are the logs of the app:
```
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/lib/x86_64-linux-gnu/libcurl-gnutls.so.4: undefined symbol: nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgiolibproxy.so
/usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so: undefined symbol: g_assertion_message_cmpint
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
-- app starts up --

-- click on logs and freeze --
```

some errors while launching but no logs after that

**Comment by Norbiros** [comment]
This is some weird problem with webkit gtk. We can't really fix it easily, so we have to switch to different virtual list library.

**Comment by mcredstoner** [comment]
> Same here, it will just hang there and do nothing, ive tried several ways of killing it but nothing seems to work, other than restarting.
> 
> Linux Mint 21.3 Cinnamon ![prewview](https://private-user-images.githubusercontent.com/88260244/340059323-849fdbc6-60a6-424f-b3d1-2cba09624540.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxNDQyOTcsIm5iZiI6MTcyODE0Mzk5NywicGF0aCI6Ii84ODI2MDI0NC8zNDAwNTkzMjMtODQ5ZmRiYzYtNjBhNi00MjRmLWIzZDEtMmNiYTA5NjI0NTQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDE1NTk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2Mzc5NjViODM2YTliNGFhNDliNWM1ZTc0MGFmMjU2NWVmMDgzZTA4N2FhMGRhYmJhOTRiODI4NGUwNWEyZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1z_1H6xTdCvVcfKbsNWTBRnmL5XzOJqJ6s2aW1mCnRQ)

for that you can use an extra app called Resources
( https://flathub.org/apps/net.nokyan.Resources )

**Comment by kawazoe** [comment]
> This is some weird problem with webkit gtk. We can't really fix it easily, so we have to switch to different virtual list library.

@Norbiros Maybe a somewhat related issue, on my install (Garuda KDE 6.2) the app scrolling is insanely slow. I get maybe 5-10 FPS. Everything else works fine.

**Comment by LapisPhoenix** [comment]
> > This is some weird problem with webkit gtk. We can't really fix it easily, so we have to switch to different virtual list library.
> 
> [@Norbiros](https://github.com/Norbiros) Maybe a somewhat related issue, on my install (Garuda KDE 6.2) the app scrolling is insanely slow. I get maybe 5-10 FPS. Everything else works fine outside of the log crash. It could be that the list virtualisation is causing this too.

I have the same issue, **extremely** slow scrolling, on all menus.

**Comment by Norbiros** [comment]
Yeah, it's also webkit problem. It is generalny very slow but in AppImages this problem is even bigger. For me manually building the app (`pnpm app:build`) and using binary improves it.

**Comment by Norbiros** [comment]
On new version (`0.9.0`) setting `WEBKIT_DISABLE_DMABUF_RENDERER=0` improves performance & fixes this issue.

**Comment by kawazoe** [comment]
> On new version (`0.9.0`) setting `WEBKIT_DISABLE_DMABUF_RENDERER=0` on wayland improves performance & fixes this issue.

I just tried 0.9 and... it's been a nothing short of a disastrous experience for me*. The official app image stays stuck on a gray window and the native decorations don't do anything.

I tried 0.9.0-1 [from the AUR](https://aur.archlinux.org/packages/modrinth-app) and that one launches but, even with the envar you're suggesting, it's more meaningful to talk about the performance in _seconds per frame_ rather than frames per second. Simple animations like button hovers takes 1 to 2 seconds to reach their steady state. Resizing the window feels like I'm back on Windows XP. Even the native window decorations don't respond properly.

All of those simple actions are using 100% of a core. Even if some of this stuff should clearly be offloaded to the GPU, hovering a button should not be using that much resources. I can run a browser without hardware acceleration and watch videos with a better frame-rate so it's clear that there's something more involved going on here.

If this helps, I don't just get stalls; I get actual seg faults in libwebkit2gtk. If you have a way for me to share the coredump privately, I don't mind sending it over.

<details><summary>* Here's my setup to give you some context</summary>
<p>

 OS Garuda Linux x86_64
└󰍹 Kernel Linux 6.12.4-zen1-1-zen

 DE KDE Plasma 6.2.4
└󰍹 Window Manager KWin (Wayland)

󰌢 PC Desktop
├󰻠 CPU AMD Ryzen 9 7950X (32) @ 5.88 GHz
├󰍛 GPU AMD Radeon RX 7900 XTX [Discrete]
├󰍛 GPU AMD Raphael [Integrated]
├󰍛 OpenGL 4.6 (Compatibility Profile) Mesa 24.3.2-arch1.1
├󰍛 Vulkan 1.3.296 - radv [Mesa 24.3.2-arch1.1]
└󰍹 Display(s) 3840x2160 @ 60 Hz (as 3072x1728) in 27" [External] *

</p>
</details>

**Comment by GDudeUnderscore** [comment]
gg
